src/main/java/io/aiven/kafka/connect/http/sender/AbstractHttpSender.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2023 Aiven Oy and http-connector-for-apache-kafka project contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.http.sender;
+
+import java.io.IOException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.Builder;
+import java.net.http.HttpResponse;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.kafka.connect.errors.ConnectException;
+
+import io.aiven.kafka.connect.http.config.HttpSinkConfig;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+abstract class AbstractHttpSender {
+
+    private static final Logger log = LoggerFactory.getLogger(AbstractHttpSender.class);
+
+    protected final HttpClient httpClient;
+    protected final HttpSinkConfig config;
+    protected final HttpRequestBuilder httpRequestBuilder;
+
+    protected AbstractHttpSender(
+            final HttpSinkConfig config, final HttpRequestBuilder httpRequestBuilder, final HttpClient httpClient
+    ) {
+        Objects.requireNonNull(config);
+        this.config = config;
+        this.httpRequestBuilder = httpRequestBuilder;
+        this.httpClient = httpClient;
```

Comment:
```suggestion
        this.config = Objects.requireNonNull(config);;
        this.httpRequestBuilder = Objects.requireNonNull(httpRequestBuilder);
        this.httpClient = Objects.requireNonNull(httpClient);
```

This will make two tests fail on `OAuth2HttpSenderTest`, `mockedClient` can be used instead of `null`. And assert could be added that client is not called on those test.